Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java

```diff
@@ -14,14 +16,22 @@
  */
 public class PayMoreToCastAsThoughtItHadFlashAbility extends SpellAbility {
 
-    private final ManaCosts costsToAdd;
+    private final Cost costsToAdd;
 
     public PayMoreToCastAsThoughtItHadFlashAbility(Card card, ManaCosts<ManaCost> costsToAdd) {
-        super(card.getSpellAbility().getManaCosts().copy(), card.getName() + " as though it had flash", Zone.HAND, SpellAbilityType.BASE_ALTERNATE);
+        this(card, (Cost) costsToAdd);
+        CardUtil.increaseCost(this, costsToAdd);
+    }
+
+    public PayMoreToCastAsThoughtItHadFlashAbility(Card card, Costs<Cost> costsToAdd) {
+        this(card, (Cost) costsToAdd);
+        this.addCost(costsToAdd);
+    }
+
+    public PayMoreToCastAsThoughtItHadFlashAbility(Card card, Cost costsToAdd) {
+        super(card.getSpellAbility().getManaCosts().copy(), card.getName(), Zone.HAND, SpellAbilityType.BASE_ALTERNATE);
         this.costsToAdd = costsToAdd;
         this.timing = TimingRule.INSTANT;
-        this.setRuleAtTheTop(true);
```

Comment:
why did you remove setRuleAtTheTop? always true for this ability right?

Reply:
That's how it should be, but setRuleAtTheTop does nothing in case of this ability. Its rule text is always at the bottom in generated rules despite of card image text. I didn't research the cause and relied on previous experience, considering it right.